test/storages/dataset.test.ts

```diff
@@ -310,6 +320,7 @@ describe('dataset', () => {
             const { dataset, restoreAndVerify } = getRemoteDataset();
 
             const result = await dataset.map((item, index) => {
+                // @ts-expect-error item is unknown, casting needed
```

Comment:
Ideally we should use unknown instead of any because of the added type benefits (makes you cast your data to be sure you're doing the right thing). However, I agree with making it a generic, so I'll look into that

Reply:
looking at the code, we require the items to be objects, so it should be a `Dictionary` instead of `unknown`

https://github.com/apify/apify-ts/blob/master/packages/apify/src/storages/dataset.ts#L178